### PR TITLE
Alternate for fp->bool Cast using Greater+ABS

### PR DESCRIPTION
**Pull request by qti-shubham**
### Description
Not_Equal op has accuracy issue on v79 architecture: https://jira-dc.qualcomm.com/jira/browse/AISW-192595

All the Attention Mask Softmax block were affected.

FP->Bool Cast op is now natively supported by QNN (tested with v2.47 on device) but fails on CI (esp. Win Arm64 v73)

**This is alternate approach using Greater(Abs(x), 0) to workaround CAST(FP->Bool)**

### CODE REVIEW

File: onnxruntime/core/providers/qnn/builder/opbuilder/cast_op_builder.cc [hunk 1]
```diff
@@ -58,17 +64,56 @@ bool CastOpBuilder::IsFpToBoolCast(const OrtNodeUnit& node_unit) const {
...
+  const std::string abs_node_name = utils::UniqueNameGenerator().New(node_unit, "_abs_node");
+  RETURN_IF_ERROR(AddQnnScalar<uint32_t>(qnn_model_wrapper, node_unit.Index(), abs_node_name,
+                                         static_cast<uint32_t>(QNN_OP_ELEMENT_WISE_UNARY_OPERATION_ABS),
+                                         QNN_OP_ELEMENT_WISE_UNARY_PARAM_OPERATION, abs_param_names));
+  RETURN_IF_NOT(qnn_model_wrapper.CreateQnnNode(abs_node_name,
+                                                QNN_OP_PACKAGE_NAME_QTI_AISW,
+                                                QNN_OP_ELEMENT_WISE_UNARY,
+                                                {input_names.back()},
+                                                {abs_out_name},
+                                                std::move(abs_param_names)),
```
Comment (yath1) [inline onnxruntime/core/providers/qnn/builder/opbuilder/cast_op_builder.cc:98]: Missing do_op_validation.

**Review by yath1 — APPROVED** [review]
Minor comment otherwise looks good to me.

**Comment by qti-shubham** [comment]
image

It looks like on v79, HTP kernels doing comparison with 0 have bug.

**Comment by qti-shubham** [comment]
Abandoning this PR, as there is bug with 0 comparison on v79 HTP kernel.
Can't use this workaround. The direct Cast operator worked.
